packages/0x.js/src/utils/decorators.ts

```diff
@@ -85,7 +91,40 @@ const syncErrorHandlerFactory = (errorTransformer: ErrorTransformer) => {
 // _.flow(f, g) = f ∘ g
 const zeroExErrorTransformer = _.flow(schemaErrorTransformer, contractCallErrorTransformer);
 
+const addParameterTransformer = (target: object, key: string | symbol, transformer: ParameterTransformer) => {
+    const param_key = `parameter_transformer_${key}`;
+    const transformableParams: ParameterTransformer[] = Reflect.getOwnMetadata(param_key, target, key) || [];
+    transformableParams.push(transformer);
+    Reflect.defineMetadata(param_key, transformableParams, target, key);
+};
+
+const ethereumAddressParameterDecorator = (target: object, key: string | symbol, parameterIndex: number) => {
+    const transformer = { parameterIndex, transform: (value: string): string => value.toLowerCase() };
+    addParameterTransformer(target, key, transformer);
+};
+
+const parameterTransformerMethodDecorator = (
+    target: object,
+    key: string | symbol,
+    descriptor: TypedPropertyDescriptor<any>,
+) => {
+    const param_key = `parameter_transformer_${key}`;
+    const transformableParams: ParameterTransformer[] = Reflect.getOwnMetadata(param_key, target, key) || [];
+    // Do not use arrow syntax here. Use a function expression in
+    // order to use the correct value of `this` in this method
+    // tslint:disable-next-line:only-arrow-functions
+    descriptor.value = function(...args: any[]) {
+        _.each(transformableParams, transformer => {
+            args[transformer.parameterIndex] = transformer.transform(args[transformer.parameterIndex]);
+        });
+        return descriptor.value.apply(this, args);
+    };
+    return descriptor;
+};
+
 export const decorators = {
     asyncZeroExErrorHandler: asyncErrorHandlerFactory(zeroExErrorTransformer),
     syncZeroExErrorHandler: syncErrorHandlerFactory(zeroExErrorTransformer),
+    ethereumAddress: ethereumAddressParameterDecorator,
+    parameterTransformer: parameterTransformerMethodDecorator,
```

Comment:
`parameterTransformer` seems to have the same method signature as `asyncZeroExErrorHandler` and `syncZeroExErrorHandler`, should we call it `parameterHandler` instead?